src/pages/studyView/studyPageHeader/rightPanel/RightPanel.tsx

```diff
@@ -69,12 +68,9 @@ export default class RightPanel extends React.Component<IRightPanelProps, {}> {
                                     found: Gene[];
                                     suggestions: GeneReplacement[];
                                 },
-                                queryStr: string,
-                                status: "pending" | "error" | "complete") {
-        this._isQueryButtonDisabled = queryStr==='' || (status === 'pending') || !_.isUndefined(oql.error) || genes.suggestions.length !== 0;
-        if (status === "complete") {
-            this.props.store.updateSelectedGenes(oql.query, genes.found);
-        }
+                                queryStr: string) {
+        this.geneValidationHasIssue = queryStr==='' || !_.isUndefined(oql.error) || genes.suggestions.length !== 0;
```

Comment:
possibly not, but could this (geneValidationHasIssue) be computed?  the reason it catches my eye is that it's a state that has to be cleared, and yet depends on other observables.

Reply:
It actually only depends on the params, not observables.